mypyc/irbuild/for_helpers.py

```diff
@@ -1166,3 +1178,73 @@ def gen_step(self) -> None:
     def gen_cleanup(self) -> None:
         for gen in self.gens:
             gen.gen_cleanup()
+
+
+class ForMap(ForGenerator):
+    """Generate optimized IR for a for loop over map(f, ...)."""
+
+    def need_cleanup(self) -> bool:
+        # The wrapped for loops might need cleanup. We might generate a
+        # redundant cleanup block, but that's okay.
+        return True
+
+    def init(self, func: Expression, exprs: list[Expression]) -> None:
+        self.func_expr = func
+        self.func = self.builder.accept(func)
+        self.exprs = exprs
+        self.cond_blocks = [BasicBlock() for _ in range(len(exprs) - 1)] + [self.body_block]
+
+        self.gens: list[ForGenerator] = []
+        for i, iterable_expr in enumerate(exprs):
+            argname = f"_mypyc_map_arg_{i}"
+            var_type = self.builder._analyze_iterable_item_type(iterable_expr)
+            name_expr = NameExpr(argname)
+            name_expr.kind = LDEF
+            name_expr.node = Var(argname, var_type)
+            self.builder.add_local_reg(name_expr.node, self.builder.type_to_rtype(var_type))
+            self.gens.append(
+                make_for_loop_generator(
+                    self.builder,
+                    name_expr,
+                    iterable_expr,
+                    self.cond_blocks[i],
+                    self.loop_exit,
+                    self.line,
+                    is_async=False,
+                    nested=True,
+                )
+            )
+
+    def gen_condition(self) -> None:
+        for i, gen in enumerate(self.gens):
+            gen.gen_condition()
+            if i < len(self.gens) - 1:
+                self.builder.activate_block(self.cond_blocks[i])
+
+    def begin_body(self) -> None:
+        builder = self.builder
+        line = self.line
+
+        for gen in self.gens:
+            gen.begin_body()
+
+        # This goes here to prevent a circular import
+        from mypyc.irbuild.expression import transform_call_expr
```

Comment:
Use `builder.accept` instead of calling `transform_call_expr` directly. I think this will avoid the circular dependency.